src/admin/class-wordlift-admin-entity-list.php

```diff
@@ -80,4 +80,73 @@ public function render_custom_columns( $column, $entity_id ) {
 		}
 		
 	}
+	
+	/**
+	 * Add 4W select box before the 'Filter' button.
+	 *
+	 * @since 3.3.0
+	 */
+	public function add_4W_filter() {
+		global $typenow;
+		
+		// Only show on entity list page
+		if( $typenow !== Wordlift_Entity_Service::TYPE_NAME ){
+			return;
+		}
+		
+		// Was a W already selected?
+		$selected = isset( $_GET['4W_filter'] ) ? $_GET['4W_filter'] : '' ;
+		
+		// Build select box with the 4W
+		$all_w = array(
+			'Select W',
+			WL_WHAT_RELATION,
+			WL_WHO_RELATION,
+			WL_WHERE_RELATION,
+			WL_WHEN_RELATION
+		);
+		$output  = '<select name="4W_filter" id="dropdown_4W_type">';
```

Comment:
Do not use underscores in html attributes (name/id), use dash instead: https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Underscores_in_class_and_ID_Names